src/test/groovy/com/github/roroche/plantuml/android/AndroidProjectTypeTest.groovy

```diff
@@ -0,0 +1,44 @@
+package com.github.roroche.plantuml.android
+
+import com.github.roroche.plantuml.urls.MainOutputUrlsFactory
+import com.github.roroche.plantuml.urls.Urls
+import org.gradle.api.Project
+import org.gradle.testfixtures.ProjectBuilder
+import org.junit.jupiter.api.Test
+
+class AndroidProjectTypeTest {
+    private final Project project = ProjectBuilder.builder().build()
+
+    @Test
+    void testProjectIsAndroidApplication() {
+        project.apply plugin: "com.android.application"
+        project.android {
+            compileSdkVersion 21
+        }
+        project.evaluate()
+
+        assert AndroidProjectType.isAndroidProject(project)
+        assert AndroidProjectType.isAndroidApplication(project)
+        assert !AndroidProjectType.isAndroidLibrary(project)
```

Comment:
These testing concerns must go to a dedicated `Assertion` implementation. Here is why: https://www.pragmaticobjects.com/chapters/003_reusable_assertions.html

The project is based on this library: https://github.com/pragmatic-objects/oo-tests

Here is an example: https://github.com/RoRoche/plantuml-gradle-plugin/blob/master/src/test/groovy/com/github/roroche/plantuml/BuildClassDiagramTaskTest.groovy

Reply:
But maybe these tests are not worth considering now and we should focus the effort on the next ones (those about Urls generation)